src/server/editorServices.ts

```diff
@@ -851,21 +854,31 @@ namespace ts.server {
             return { success: true, projectOptions, configFileErrors: errors };
         }
 
-        private exceededTotalSizeLimitForNonTsFiles<T>(options: CompilerOptions, fileNames: T[], propertyReader: FilePropertyReader<T>) {
+        private exceededTotalSizeLimitForNonTsFiles<T>(name: string, options: CompilerOptions, fileNames: T[], propertyReader: FilePropertyReader<T>) {
             if (options && options.disableSizeLimit || !this.host.getFileSize) {
                 return false;
             }
+
+            let availableSpace = maxProgramSizeForNonTsFiles;
+            this.projectToSizeMap[name] = 0;
+            for (const key in this.projectToSizeMap) {
+                availableSpace -= (this.projectToSizeMap[key] || 0);
+            }
+
             let totalNonTsFileSize = 0;
             for (const f of fileNames) {
                 const fileName = propertyReader.getFileName(f);
                 if (hasTypeScriptFileExtension(fileName)) {
                     continue;
                 }
                 totalNonTsFileSize += this.host.getFileSize(fileName);
-                if (totalNonTsFileSize > maxProgramSizeForNonTsFiles) {
+                if (totalNonTsFileSize > availableSpace) {
+                    this.projectToSizeMap[name] = totalNonTsFileSize;

```

Comment:
Do we disable the language service for A, B, and C regardless (i.e. the entire solution), or only for the project where the "exceeds" check fails? The way this is written, it will fail for every project checked after "B", as you still add its file size to the point of failure to the map - even though that project won't load. If you didn't do that, you could still load C after B fails to load (as A & C combined are only 2MB).

Make sense? Feel free to ping me on IM if not.